### DISCUSSION TITLE
Refactoring Node2ObjectsMap, Node2PipelineMap, Node2ContainerMap

### DISCUSSION THREAD

**Discussion by Montura** [apache/ozone#6294] [Ideas]
There is a base class [`org.apache.hadoop.hdds.scm.node.states.Node2ObjectsMap`](https://github.com/apache/ozone/blob/master/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/states/Node2ObjectsMap.java#L40) for [`Node2PipelineMap`](https://github.com/apache/ozone/blob/master/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/states/Node2PipelineMap.java#L37) and [`Node2ContainerMap`](https://github.com/apache/ozone/blob/master/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/states/Node2ContainerMap.java#L37).

It is redundant because of:
1. `Node2ContainerMap` is used only in tests
2. `Node2PipelineMap` doesn't  need inheritance at all

Moreover `Node2PipelineMap` contains `ConcurrentHashMap<UUID, Set<PipelineID>>`and 2 `synchronized` methods:
* `void addPipeline(Pipeline pipeline)` -> [compute if absent for concurrent hash map](https://github.com/apache/ozone/blob/master/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/states/Node2PipelineMap.java#L68)
* `void removePipeline(Pipeline pipeline)` -> [compute if present for concurrent hash map](https://github.com/apache/ozone/blob/master/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/states/Node2PipelineMap.java#L76)

I suppose `synchronized` is redundant here for these two methods.

There is a proposed [change](https://github.com/apache/ozone/commit/edda28f3b41f50924255f7f651e4c3dafb3b8831):
1. Remove `org.apache.hadoop.hdds.scm.node.states.Node2ObjectsMap` class
2. Move `Node2ContainerMap` to tests
3. Don't use `synchronized` for `Node2PipelineMap::addPipeline` and `Node2PipelineMap::removePipeline`

**Comment by kerneltime** [comment]
Ref: https://github.com/apache/ozone/pull/6303
